domain/src/main/java/edu/nextstep/camp/calculator/domain/ExpressionParser.kt

```diff
@@ -0,0 +1,38 @@
+package edu.nextstep.camp.calculator.domain
+
+class ExpressionParser {
+
+    fun parse(expression: String): Expression {
+        val signList = findSigns(expression)
+        val numberList = findNumbers(expression)
+
+        return Expression(
+            numberList = numberList,
+            signList = signList
+        )
+    }
+
+    private fun findSigns(expression: String): List<Sign> {
+        return """[^0-9 ]""".toRegex()
+            .findAll(expression)
+            .map {
+                when (it.value) {
+                    "+" -> Sign.PLUS
+                    "-" -> Sign.MINUS
+                    "*" -> Sign.TIMES
+                    "/" -> Sign.DIVISION
+                    else -> throw IllegalArgumentException("수식에는 +, -, *, /만 사용될 수 있습니다.")
+                }
+            }
+            .toList()
+    }
```

Comment:
제가 생각하는 선언적 함수형 코드는 위에서 아래로 읽기가 편함에 그 장점이 있다고 생각합니다. 하지만 각 함수 chain 안에 많은 코드가 들어가게 되면 그 장점이 퇴색되는 것 같습니다. 

```suggestion
    private fun findSigns(expression: String): List<Sign> {
        return """[^0-9 ]""".toRegex()
            .findAll(expression)
            .map(::findMappedSign)
            .toList()
    }

    private fun findMappedSign(it: MatchResult) = when (it.value) {
        "+" -> Sign.PLUS
        "-" -> Sign.MINUS
        "*" -> Sign.TIMES
        "/" -> Sign.DIVISION
        else -> throw IllegalArgumentException("수식에는 +, -, *, /만 사용될 수 있습니다.")
    }
```

함수 chain안에 많은 양의 코드가 있는 경우 함수로 분리해서 선언적 함수형 코드의 장점을 살리면 좋을 것 같습니다. 😘